fastscanner/services/indicators/lib/candle.py

```diff
@@ -255,6 +255,69 @@ def column_name(self):
     def lookback_days(self) -> int:
         return 0
 
+    async def extend(self, symbol: str, df: pd.DataFrame) -> pd.DataFrame:
+        df = await self._high_n_days.extend(symbol, df)
+        df = await self._low_n_days.extend(symbol, df)
+
+        high_col = self._high_n_days.column_name()
+        low_col = self._low_n_days.column_name()
+
+        df[self.column_name()] = (df[CandleCol.CLOSE] - df[low_col]) / (
+            df[high_col] - df[low_col]
+        )
+
+        return df.drop(columns=[high_col, low_col])
+
+    async def extend_realtime(self, symbol: str, new_row: pd.Series) -> pd.Series:
+        new_row = await self._high_n_days.extend_realtime(symbol, new_row)
+        new_row = await self._low_n_days.extend_realtime(symbol, new_row)
+
+        high_col = self._high_n_days.column_name()
+        low_col = self._low_n_days.column_name()
+
+        try:
+            high = new_row[high_col]
+            low = new_row[low_col]
+            close = new_row[CandleCol.CLOSE]
+
+            if pd.isna(high) or pd.isna(low) or high == low:
+                new_row[self.column_name()] = pd.NA
+            else:
+                new_row[self.column_name()] = (close - low) / (high - low)
+        except KeyError:
+            new_row[self.column_name()] = pd.NA
+
+        return new_row
```

Comment:
You need to drop the columns, but only if they didn't exist in the original new_row already.